source/frontend/src/features/mapSideBar/disposition/models/DispositionSaleFormModel.ts

```diff
@@ -1,35 +1,49 @@
-import { Api_DispositionFileSale } from '@/models/api/DispositionFile';
+import { Api_DispositionFileSale, Api_DispositionSaleContact } from '@/models/api/DispositionFile';
 import { emptyStringtoNullable } from '@/utils/formUtils';
 
-export class DispositionSaleFormModel {
+import { DispositionSaleContactModel, WithSalePurchasers } from './DispositionSaleContactModel';
+
+export class DispositionSaleFormModel implements WithSalePurchasers {
   finalConditionRemovalDate: string | null = null;
   saleCompletionDate: string | null = null;
   saleFiscalYear: string | null = null;
   finalSaleAmount: number | null = null;
   realtorCommissionAmount: number | null = null;
-  isGstRequired: boolean | null = null;
+  isGstRequired: boolean = false;
   gstCollectedAmount: number | null = null;
   netBookAmount: number | null = null;
   totalCostAmount: number | null = null;
   netProceedsBeforeSppAmount: number | null = null;
   sppAmount: number | null = null;
   netProceedsAfterSppAmount: number | null = null;
   remediationAmount: number | null = null;
+  dispositionPurchasers: DispositionSaleContactModel[] = [];
+  dispositionPurchaserAgent: DispositionSaleContactModel = new DispositionSaleContactModel();
+  dispositionPurchaserSolicitor: DispositionSaleContactModel = new DispositionSaleContactModel();
 
-  constructor(readonly id: number | null = null, readonly dispositionFileId: number) {
+  constructor(
+    readonly id: number | null = null,
+    readonly dispositionFileId: number,
+    readonly rowVersion: number | null = null,
+  ) {
     this.id = id;
     this.dispositionFileId = dispositionFileId;
+    this.rowVersion = rowVersion;
   }
 
-  static fromApi(entity: Api_DispositionFileSale) {
-    const model = new DispositionSaleFormModel(entity.id, entity.dispositionFileId);
+  static fromApi(entity: Api_DispositionFileSale): DispositionSaleFormModel {
```

Comment:
This is a pretty complex model, it looks like a good candidate for model unit tests. See some of our other model unit tests for example (they are quite simple).

Reply:
updated